Framework/DataHandling/src/LoadNGEM.cpp

```diff
@@ -112,11 +112,10 @@ void createEventWorkspace(const int &maxToF, const double &binWidth,
   dataWorkspace = DataObjects::create<DataObjects::EventWorkspace>(
       NUM_OF_SPECTRA, HistogramData::Histogram(HistogramData::BinEdges(xAxis)));
   PARALLEL_FOR_NO_WSP_CHECK()
-  for (size_t i = 0; i < events.size(); ++i) {
+  for (int i = 0; i < events.size(); ++i) {
```

Comment:
Thanks for fixing this. The current build problems are due to the Linux compilers producing a warning that there is a unsigned compared with a signed number in `i < events.size()` as `.size` returns an unsigned.

This basically forces us to always write these loops as:

```
  for (int64_t i = 0; i < static_cast<int64_t>(events.size()); ++i) {
```

We cast the size to `int64` to make sure that we don't cut off the size if it is over what would fit in a 32-bit `int`.

Reply:
OK thanks. The index i is subsequently used to set the spectrum number and detector number in the workspace (setSpectrumNo and setDetectorID) so it will have to end up as a 32 bit int. 

Looking a bit further around that unit the events vector and the workspace that is being created are both set up to be of size NUM_OF_SPECTRA which is an int constant. I will change the for loop to iterate up to NUM_OF_SPECTRA and set the for loop index to be int. Please let me know if that looks better once I've done the change

Reply:
I've pushed the change I described earlier and all the checks pass now apart from "Ubuntu + Doc Tests" which fails on this (which isn't obviously related to me):
LiveDataTest_KafkaHistoStreamDecoderTest

13:26:42 Running 1 testFacilityInfo-[Notice] No timezone specified for TEST in Facilities.xml
13:26:42 
13:26:42 In KafkaHistoStreamDecoderTest::test_Histo_Stream:
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:80: Error: Test failed: Decoder's data buffers should be created now
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:80: Error: Assertion failed: decoder->hasData()
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:81: Error: Test failed: No message to process yet.
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:81: Error: Expected (workspace = decoder->extractData()) not to throw, but it did
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:87: Error: Test failed: Expected non-null workspace pointer from extractData()
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:87: Error: Assertion failed: workspace
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:91: Error: Test failed: Expected a Workspace2D from extractData(). Found something else
13:26:42 /jenkins_workdir/workspace/pull_requests-ubuntu/Framework/LiveData/test/KafkaHistoStreamDecoderTest.h:91: Error: Assertion failed: histoWksp
13:26:42 Segmentation fault

Reply:
Good point about the maximum size being fixed. I'd forgotten this was for a very specific detector configuration.

As you may have noted already as the job passes now we have a couple of tests in the system that seem to have some instability in them. They always run find on developers machines on their own but occasionally when run in parallel on builders they fail.